services/server/src/server/services/utils/database-util.ts

```diff
@@ -858,121 +864,86 @@ export async function getDatabaseColumnsFromVerification(
-    );
-  }
-
-  try {
-    // Fetch compilation data from the database
-    const verifiedContractResult =
-      await database.getSourcifyMatchByChainAddressWithProperties(
-        chainId,
-        bytesFromString(address),
-        [
-          "std_json_input",
-          "std_json_output",
-          "runtime_cbor_auxdata",
-          "creation_cbor_auxdata",
-          "fully_qualified_name",
-          "version",
-          "metadata",
-        ],
-      );
+export function prepareCompilerSettingsFromJsonInput(
+  jsonInput: SolidityJsonInput | VyperJsonInput,
+): CompilerSettings {
+  return removeOutputSelectionFromSettings(jsonInput.settings);
+}
 
-    if (verifiedContractResult.rows.length === 0) {
-      logger.error(
-        "extractCompilationFromDatabase: verified contract not found",
-        {
-          chainId,
-          address,
-        },
-      );
-      throw new Error("Verified contract not found");
-    }
+export function prepareCompilerSettingsFromVerification(
+  verification: VerificationExport,
+): CompilerSettings {
+  return removeOutputSelectionFromSettings(
+    verification.compilation.jsonInput.settings,
+  );
+}
 
-    const verifiedContract = verifiedContractResult.rows[0];
-
-    // Extract properties from the verified contract
-    const compilerVersion = verifiedContract.version;
-    const creationCodeCborAuxdata: CompiledContractCborAuxdata | undefined =
-      verifiedContract.creation_cbor_auxdata || undefined;
-    const runtimeCodeCborAuxdata: CompiledContractCborAuxdata | undefined =
-      verifiedContract.runtime_cbor_auxdata || undefined;
-
-    // Get the file path and contract name from fully_qualified_name
-    const { contractPath, contractName } = splitFullyQualifiedName(
-      verifiedContract.fully_qualified_name!,
-    );
-    const compilationTarget = {
-      name: contractName,
-      path: contractPath,
-    };
+export function createPreRunCompilationFromStoredCandidate(
+  { solc, vyper }: { solc: ISolidityCompiler; vyper: IVyperCompiler },
+  candidate: SimilarityCandidate,
+): PreRunCompilation {
+  const {
+    std_json_input: jsonInput,
+    std_json_output: jsonOutput,
+    version,
+    fully_qualified_name: fullyQualifiedName,
+    creation_cbor_auxdata: creationCborAuxdata,
+    runtime_cbor_auxdata: runtimeCborAuxdata,
+    metadata,
+  } = candidate;
+
+  if (
+    !jsonInput ||
+    !jsonOutput ||
+    !version ||
+    !fullyQualifiedName ||
+    !creationCborAuxdata ||
+    !runtimeCborAuxdata
+  ) {
+    logger.error("createPreRunCompilationFromStoredCandidate: missing fields", {
+      hasJsonInput: !!jsonInput,
+      hasJsonOutput: !!jsonOutput,
+      hasVersion: !!version,
+      hasFullyQualifiedName: !!fullyQualifiedName,
+      hasCreationCborAuxdata: !!creationCborAuxdata,
+      hasRuntimeCborAuxdata: !!runtimeCborAuxdata,
+    });
+    throw new Error("Compilation properties not found");
+  }
```

Comment:
you don't need all of these checks anymore. The point of introducing `SimilarityCandidate` was reducing the necessity of such checks.